### PR TITLE
(DOCS) Update help for puppet-device

### PR DESCRIPTION
**Pull request by tkishel**
Prior to this commit, the help for puppet-device ...

  contained language and syntax errors

  did not document all of the available device.conf properties

  included specific device configuration that should be documented in device.conf

See also: https://github.com/puppetlabs/puppet-docs/pull/756
Refer to (DOC-1967) for more information.

### CODE REVIEW

File: lib/puppet/application/device.rb [hunk 1]
```diff
@@ -70,51 +70,54 @@ def help
 
 SYNOPSIS
 --------
-Retrieves all configurations from the puppet master and apply
-them to the remote devices configured in /etc/puppetlabs/puppet/device.conf.
+Retrieves catalogs from the puppet master and applies them to remote devices.
```
Comment (jtappa) [inline lib/puppet/application/device.rb]: Puppet should be capitalized

File: lib/puppet/application/device.rb [hunk 2]
```diff
@@ -70,51 +70,54 @@ def help
 
 SYNOPSIS
 --------
-Retrieves all configurations from the puppet master and apply
-them to the remote devices configured in /etc/puppetlabs/puppet/device.conf.
+Retrieves catalogs from the puppet master and applies them to remote devices. 
+
+This subcommand may be run manually; or periodically using cron,
```
Comment (jtappa) [inline lib/puppet/application/device.rb]: "can" instead of "may"

File: lib/puppet/application/device.rb [hunk 3]
```diff
@@ -70,51 +70,54 @@ def help
...
                 [-v|--verbose] [-w|--waitforcert <seconds>]
-                [-t|--target <device>] [-V|--version]
+                [-t|--target <device>] [--user=<user>] [-V|--version]
 
 
 DESCRIPTION
 -----------
-Once the client has a signed certificate for a given remote device, it will
-retrieve its configuration and apply it.
+Devices require a proxy puppet agent to request certificates, collect facts,
```
Comment (jtappa) [inline lib/puppet/application/device.rb]: Capitalize Puppet here

File: lib/puppet/application/device.rb [hunk 4]
```diff
@@ -70,51 +70,54 @@ def help
...
+retrieve and apply catalogs, and store reports.
+
 
 USAGE NOTES
 -----------
-One need a /etc/puppetlabs/puppet/device.conf file with the following content:
+Devices managed by the puppet-device subcommand on a puppet agent are
```
Comment (jtappa) [inline lib/puppet/application/device.rb]: Capitalize Puppet here

File: lib/puppet/application/device.rb [hunk 5]
```diff
@@ -70,51 +70,54 @@ def help
...
-retrieve its configuration and apply it.
+Devices require a proxy puppet agent to request certificates, collect facts,
+retrieve and apply catalogs, and store reports.
+
 
 USAGE NOTES
 -----------
-One need a /etc/puppetlabs/puppet/device.conf file with the following content:
+Devices managed by the puppet-device subcommand on a puppet agent are 
+configured in device.conf, located at $confdir/device.conf by default,
```
Comment (jtappa) [inline lib/puppet/application/device.rb]: needs "which is" added:

`configured in device.conf, which is located at $confdir/device.conf by default,`

**Comment by tkishel** [comment]
Thanks. I've implemented all of these review changes. I have PRs targeting the 4.10.x and master branches: the content differs (but I've applied these review changes to both). I can close the PR against the master branch, if necessary.

**Comment by jtappa** [comment]
So, as I understand it you just need to make the PR on one branch, and our merge process will bring it forward automatically to the other versions. @joshcooper correct me if I have this jumbled up?

**Comment by MosesMendoza** [comment]
Right ideally the workflow is we merge the 4.10.x content to 4.10.x, and then merge that change to master. Then a second PR is raised that modifies the (now 4.10.x) content in master with the master-only changes.

**Comment by tkishel** [comment]
I have the 4.10.x PR at: https://github.com/puppetlabs/puppet/pull/6012

Shall I close this one?

**Comment by joshcooper** [comment]
Superseded by #6012

**Pull request closed without merging**